src/main/java/org/minions/devfund/noemiguzman/battleship/Ship.java

```diff
@@ -0,0 +1,188 @@
+package org.minions.devfund.noemiguzman.battleship;
+
+/**
+ * Ship class.
+ */
+public abstract class Ship {
+    private int length;
+    private int bowRow;
+    private int bowColumn;
+    private boolean horizontal;
+    private boolean[] hit;
+
+    /**
+     * This is an abstract method and therefore has no body.
+     *
+     * @return ship string
+     */
+    public abstract String getShipType();
+
+    /**
+     * Returns true if it is okay to put a ship of this length with its bow in this location,
+     * with the given orientation, and returns false otherwise.
+     * The ship must not overlap another ship, or touch another ship (vertically, horizontally, or diagonally),
+     * and it must not stickout beyond the array.
+     * Do not actually change either the ship or the Ocean, just says whether it is legal to do so.
+     *
+     * @param row        numeric
+     * @param column     numeric
+     * @param horizontal true
+     * @param ocean      object
+     * @return true
+     */
+    public boolean okToPlaceShipAt(int row, int column, boolean horizontal, final Ocean ocean) {
+
+
+        if (horizontal) {
+            if (column + length > ocean.SIZE_OCEAN) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this instance-reference to a static reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2209)